vortex-datafusion/src/lib.rs

```diff
@@ -94,108 +127,297 @@ impl TableProvider for VortexInMemoryTableProvider {
     /// The array is flattened directly into the nearest Arrow-compatible encoding.
     async fn scan(
         &self,
-        _state: &SessionState,
+        state: &SessionState,
         projection: Option<&Vec<usize>>,
         filters: &[Expr],
         _limit: Option<usize>,
     ) -> DFResult<Arc<dyn ExecutionPlan>> {
-        if !filters.is_empty() {
-            return exec_err!("vortex does not support filter pushdown");
+        fn get_filter_projection(exprs: &[Expr], schema: SchemaRef) -> Vec<usize> {
+            let referenced_columns: HashSet<String> =
+                exprs.iter().flat_map(get_column_references).collect();
+
+            let projection: Vec<usize> = referenced_columns
+                .iter()
+                .map(|col_name| schema.column_with_name(col_name).unwrap().0)
+                .sorted()
+                .collect();
+
+            projection
         }
 
+        let filter_exprs: Option<Vec<Expr>> = if filters.is_empty() {
+            None
+        } else {
+            Some(filters.to_vec())
```

Comment:
my earlier comment meant that you can get rid of this `to_vec`

Reply:
done